src/main/java/core/gui/theme/ImageUtilities.java

```diff
@@ -648,6 +654,31 @@ public static Icon getSvgIcon(String key, int width, int height) {
         return IconLoader.get().getIcon(Objects.requireNonNull(imagePath).toString(), width, height);
     }
 
+	/**
+	 * Transforms an icon into an image.
+	 * Cf. 	https://stackoverflow.com/a/5831357
+	 * @param icon
+	 * @return
+	 */
+	public static Image iconToImage(Icon icon) {
+		if (icon instanceof ImageIcon) {
+			return ((ImageIcon)icon).getImage();
+		}
+		else {
+			int w = icon.getIconWidth();
+			int h = icon.getIconHeight();
+			GraphicsEnvironment ge =
+					GraphicsEnvironment.getLocalGraphicsEnvironment();
+			GraphicsDevice gd = ge.getDefaultScreenDevice();
+			GraphicsConfiguration gc = gd.getDefaultConfiguration();
+			BufferedImage image = gc.createCompatibleImage(w, h);
```

Comment:
This should probably be more explicit with the type of image to create i.e. `Transparency.BITMASK` (as icons could be transparent).